### PR TITLE
fix(video): preserve like counts and engagement tags after metadata edit

### LINKED ISSUE #3923: fix: likes dissappeared and repost went from 85 to zero.
### Summary

----------------------------------------------

Dionne Maria, May 4, 2026, 04:36

likes dissappeared and repost went from 85 to zero.

my likes arent showing since I changed the hashtags, also hashtags dissappearnsometimes and my repost all of the sudden went to 85 on the first video back to 0

App Version: 1.0.9+673

### Device Information
- **platform:** ios
- **model:** iPhone
- **systemName:** iOS
- **systemVersion:** 26.3.1
- **name:** iPhone

**Current Screen:** SupportCenterScreen
**User Pubkey:** npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h

### Recent Logs (errors/warnings + recent context)
```
[2026-05-03T18:34:06.397940] [WARNING] [VIDEO] (VideoEventService) Cannot subscribe - Nostr service not initialized (will retry when ready)
[2026-05-03T18:34:06.500120] [WARNING] [SYSTEM] Zendesk JWT: SDK not initialized
[2026-05-03T18:34:06.521977] [ERROR] [SYSTEM] (PushNotificationService) NIP-44 encryption failed for token registration
[2026-05-03T18:34:07.909749] [WARNING] [VIDEO] (VideoEventService) Default video system is disabled - users will see real content only
[2026-05-03T18:34:17.919394] [DEBUG] [VIDEO] (PooledPlayer) session_summary index=3 videoId=093c29992ef45b46f356b3a0bf2e249a6eeaa2eb4bf4ede87a1d9ae6d245d7a4 url=https://media.divine.video/83514e6f3969473e9b304e6616dc6ae3296d885ad785f9c2e6b62c9cee6d5534/720p.mp4 reason=dispose finalSource=https://media.divine.video/83514e6f3969473e9b304e6616dc6ae3296d885ad785f9c2e6b62c9cee6d5534/720p.mp4 finalSourceKind=progressive loadFailovers=0 staleRecoveries=0 staleEscalations=0 stuckFailovers=0 errorType=nu... [truncated]
[2026-05-03T18:34:18.141710] [DEBUG] [RELAY] (ProfileRepository.fetchFreshProfile) Fetched from indexer relay: Dionne Maria
[2026-05-03T18:34:18.898525] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/settings, authState=authenticated
[2026-05-03T18:34:18.898700] [INFO] [SYSTEM] (Route) CTX derive: type=RouteType.settings npub=null index=null
[2026-05-03T18:34:18.898737] [INFO] [VIDEO] (VideoControllerCleanup) 🧹 Route type changed (RouteType.profile → RouteType.settings), disposing all video controllers
[2026-05-03T18:34:18.898741] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:18.914405] [INFO] (ScreenAnalytics) 📱 Screen load started: settings
[2026-05-03T18:34:18.914407] [DEBUG] [UI] (PageLoadObserver) Page push tracked: settings
[2026-05-03T18:34:18.914418] [DEBUG] [VIDEO] (VideoControllerCleanup) 🛡️ Skipping duplicate invalidation (15ms since last)
[2026-05-03T18:34:18.914421] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: settings - stopped all videos
[2026-05-03T18:34:18.916911] [DEBUG] [SYSTEM] (Nip98AuthService) 📱 Creating new NIP-98 auth token for GET https://invite.divine.video/v1/invite-status
[2026-05-03T18:34:18.916963] [INFO] [AUTH] (AuthService) Signing kind 27235 via KeycastNostrIdentity (authSource=divineOAuth, eventPubkey=698bab648010feecc9a93f233837cbfc4e95d4f049571bfa253b8ec43fdf5ef5)
[2026-05-03T18:34:18.921949] [INFO] (ProfileScreenRouter) 🧭 ProfileScreenRouter.build
[2026-05-03T18:34:18.940474] [INFO] (ScreenAnalytics) ✅ Screen content visible: settings in 26ms
[2026-05-03T18:34:18.940508] [INFO] (PagePerf) PERF: settings — visible: 26ms
[2026-05-03T18:34:18.940971] [INFO] [AUTH] (AuthService) getKnownAccounts: raw=190 chars
[2026-05-03T18:34:19.192113] [INFO] [AUTH] (AuthService) Event signed and validated: d89fdc88e47ecc12ebdd602e8a0053ca843b477bb9c3a04a065e6c805101d254
[2026-05-03T18:34:19.192168] [INFO] [SYSTEM] (Nip98AuthService) Created NIP-98 token (expires: 2026-05-03 18:44:19.192153)
[2026-05-03T18:34:19.192169] [DEBUG] [SYSTEM] (Nip98AuthService) 📱 Event ID: note1mz0aez8y0mxp967avqhg5qzne2zrk3mmh8p6qjsxtekgq5gp6f2qelu83q
[2026-05-03T18:34:19.430922] [DEBUG] [SYSTEM] (RelayNotificationsProvider) RelayNotifications: Pausing auto-refresh timer (no listeners)
[2026-05-03T18:34:19.431281] [INFO] (ProfileScreenRouter) 🧭 ProfileScreenRouter.build
[2026-05-03T18:34:22.842829] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/nostr-settings, authState=authenticated
[2026-05-03T18:34:22.843064] [INFO] [SYSTEM] (Route) CTX derive: type=RouteType.nostrSettings npub=null index=null
[2026-05-03T18:34:22.858487] [INFO] (ScreenAnalytics) 📱 Screen load started: nostr-settings
[2026-05-03T18:34:22.858489] [DEBUG] [UI] (PageLoadObserver) Page push tracked: nostr-settings
[2026-05-03T18:34:22.858496] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:22.858506] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: nostr-settings - stopped all videos
[2026-05-03T18:34:22.869677] [INFO] (ScreenAnalytics) ✅ Screen content visible: nostr-settings in 11ms
[2026-05-03T18:34:22.869713] [INFO] (PagePerf) PERF: nostr-settings — visible: 11ms
[2026-05-03T18:34:24.190826] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/key-management, authState=authenticated
[2026-05-03T18:34:24.206681] [INFO] (ScreenAnalytics) 📱 Screen load started: key-management
[2026-05-03T18:34:24.206684] [DEBUG] [UI] (PageLoadObserver) Page push tracked: key-management
[2026-05-03T18:34:24.206690] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:24.206695] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: key-management - stopped all videos
[2026-05-03T18:34:24.219184] [INFO] (ScreenAnalytics) ✅ Screen content visible: key-management in 12ms
[2026-05-03T18:34:24.219212] [INFO] (PagePerf) PERF: key-management — visible: 12ms
[2026-05-03T18:34:26.272267] [WARNING] [AUTH] (AuthService) Exporting nsec for AuthenticationSource.divineOAuth not supported
[2026-05-03T18:34:28.834035] [DEBUG] [UI] (PageLoadObserver) Page pop tracked: key-management
[2026-05-03T18:34:28.834242] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/profile/npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h, authState=authenticated
[2026-05-03T18:34:30.798375] [DEBUG] [UI] (PageLoadObserver) Page pop tracked: nostr-settings
[2026-05-03T18:34:30.798593] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/profile/npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h, authState=authenticated
[2026-05-03T18:34:31.979681] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/support-center, authState=authenticated
[2026-05-03T18:34:31.995744] [INFO] (ScreenAnalytics) 📱 Screen load started: support-center
[2026-05-03T18:34:31.995747] [DEBUG] [UI] (PageLoadObserver) Page push tracked: support-center
[2026-05-03T18:34:31.995755] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:31.995761] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: support-center - stopped all videos
[2026-05-03T18:34:32.003858] [INFO] (ScreenAnalytics) ✅ Screen content visible: support-center in 8ms
[2026-05-03T18:34:32.003889] [INFO] (PagePerf) PERF: support-center — visible: 8ms
[2026-05-03T18:34:36.456071] [DEBUG] [SYSTEM] (VideoStopNavigatorObserver) 📱 Skipping video disposal for popup route: null
[2026-05-03T18:35:59.448128] [INFO] [SYSTEM] Collecting bug report diagnostics
```

### Environment

**Platform:** 
**Build Number:** 
**OS Version:** 

### Actual Result

----------------------------------------------

Dionne Maria, May 4, 2026, 04:36

likes dissappeared and repost went from 85 to zero.

my likes arent showing since I changed the hashtags, also hashtags dissappearnsometimes and my repost all of the sudden went to 85 on the first video back to 0

App Version: 1.0.9+673

### Device Information
- **platform:** ios
- **model:** iPhone
- **systemName:** iOS
- **systemVersion:** 26.3.1
- **name:** iPhone

**Current Screen:** SupportCenterScreen
**User Pubkey:** npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h

### Recent Logs (errors/warnings + recent context)
```
[2026-05-03T18:34:06.397940] [WARNING] [VIDEO] (VideoEventService) Cannot subscribe - Nostr service not initialized (will retry when ready)
[2026-05-03T18:34:06.500120] [WARNING] [SYSTEM] Zendesk JWT: SDK not initialized
[2026-05-03T18:34:06.521977] [ERROR] [SYSTEM] (PushNotificationService) NIP-44 encryption failed for token registration
[2026-05-03T18:34:07.909749] [WARNING] [VIDEO] (VideoEventService) Default video system is disabled - users will see real content only
[2026-05-03T18:34:17.919394] [DEBUG] [VIDEO] (PooledPlayer) session_summary index=3 videoId=093c29992ef45b46f356b3a0bf2e249a6eeaa2eb4bf4ede87a1d9ae6d245d7a4 url=https://media.divine.video/83514e6f3969473e9b304e6616dc6ae3296d885ad785f9c2e6b62c9cee6d5534/720p.mp4 reason=dispose finalSource=https://media.divine.video/83514e6f3969473e9b304e6616dc6ae3296d885ad785f9c2e6b62c9cee6d5534/720p.mp4 finalSourceKind=progressive loadFailovers=0 staleRecoveries=0 staleEscalations=0 stuckFailovers=0 errorType=nu... [truncated]
[2026-05-03T18:34:18.141710] [DEBUG] [RELAY] (ProfileRepository.fetchFreshProfile) Fetched from indexer relay: Dionne Maria
[2026-05-03T18:34:18.898525] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/settings, authState=authenticated
[2026-05-03T18:34:18.898700] [INFO] [SYSTEM] (Route) CTX derive: type=RouteType.settings npub=null index=null
[2026-05-03T18:34:18.898737] [INFO] [VIDEO] (VideoControllerCleanup) 🧹 Route type changed (RouteType.profile → RouteType.settings), disposing all video controllers
[2026-05-03T18:34:18.898741] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:18.914405] [INFO] (ScreenAnalytics) 📱 Screen load started: settings
[2026-05-03T18:34:18.914407] [DEBUG] [UI] (PageLoadObserver) Page push tracked: settings
[2026-05-03T18:34:18.914418] [DEBUG] [VIDEO] (VideoControllerCleanup) 🛡️ Skipping duplicate invalidation (15ms since last)
[2026-05-03T18:34:18.914421] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: settings - stopped all videos
[2026-05-03T18:34:18.916911] [DEBUG] [SYSTEM] (Nip98AuthService) 📱 Creating new NIP-98 auth token for GET https://invite.divine.video/v1/invite-status
[2026-05-03T18:34:18.916963] [INFO] [AUTH] (AuthService) Signing kind 27235 via KeycastNostrIdentity (authSource=divineOAuth, eventPubkey=698bab648010feecc9a93f233837cbfc4e95d4f049571bfa253b8ec43fdf5ef5)
[2026-05-03T18:34:18.921949] [INFO] (ProfileScreenRouter) 🧭 ProfileScreenRouter.build
[2026-05-03T18:34:18.940474] [INFO] (ScreenAnalytics) ✅ Screen content visible: settings in 26ms
[2026-05-03T18:34:18.940508] [INFO] (PagePerf) PERF: settings — visible: 26ms
[2026-05-03T18:34:18.940971] [INFO] [AUTH] (AuthService) getKnownAccounts: raw=190 chars
[2026-05-03T18:34:19.192113] [INFO] [AUTH] (AuthService) Event signed and validated: d89fdc88e47ecc12ebdd602e8a0053ca843b477bb9c3a04a065e6c805101d254
[2026-05-03T18:34:19.192168] [INFO] [SYSTEM] (Nip98AuthService) Created NIP-98 token (expires: 2026-05-03 18:44:19.192153)
[2026-05-03T18:34:19.192169] [DEBUG] [SYSTEM] (Nip98AuthService) 📱 Event ID: note1mz0aez8y0mxp967avqhg5qzne2zrk3mmh8p6qjsxtekgq5gp6f2qelu83q
[2026-05-03T18:34:19.430922] [DEBUG] [SYSTEM] (RelayNotificationsProvider) RelayNotifications: Pausing auto-refresh timer (no listeners)
[2026-05-03T18:34:19.431281] [INFO] (ProfileScreenRouter) 🧭 ProfileScreenRouter.build
[2026-05-03T18:34:22.842829] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/nostr-settings, authState=authenticated
[2026-05-03T18:34:22.843064] [INFO] [SYSTEM] (Route) CTX derive: type=RouteType.nostrSettings npub=null index=null
[2026-05-03T18:34:22.858487] [INFO] (ScreenAnalytics) 📱 Screen load started: nostr-settings
[2026-05-03T18:34:22.858489] [DEBUG] [UI] (PageLoadObserver) Page push tracked: nostr-settings
[2026-05-03T18:34:22.858496] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:22.858506] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: nostr-settings - stopped all videos
[2026-05-03T18:34:22.869677] [INFO] (ScreenAnalytics) ✅ Screen content visible: nostr-settings in 11ms
[2026-05-03T18:34:22.869713] [INFO] (PagePerf) PERF: nostr-settings — visible: 11ms
[2026-05-03T18:34:24.190826] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/key-management, authState=authenticated
[2026-05-03T18:34:24.206681] [INFO] (ScreenAnalytics) 📱 Screen load started: key-management
[2026-05-03T18:34:24.206684] [DEBUG] [UI] (PageLoadObserver) Page push tracked: key-management
[2026-05-03T18:34:24.206690] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:24.206695] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: key-management - stopped all videos
[2026-05-03T18:34:24.219184] [INFO] (ScreenAnalytics) ✅ Screen content visible: key-management in 12ms
[2026-05-03T18:34:24.219212] [INFO] (PagePerf) PERF: key-management — visible: 12ms
[2026-05-03T18:34:26.272267] [WARNING] [AUTH] (AuthService) Exporting nsec for AuthenticationSource.divineOAuth not supported
[2026-05-03T18:34:28.834035] [DEBUG] [UI] (PageLoadObserver) Page pop tracked: key-management
[2026-05-03T18:34:28.834242] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/profile/npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h, authState=authenticated
[2026-05-03T18:34:30.798375] [DEBUG] [UI] (PageLoadObserver) Page pop tracked: nostr-settings
[2026-05-03T18:34:30.798593] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/profile/npub1dx96keyqzrlwejdf8u3nsd7tl38ft48sf9t3h7398w8vg07ltm6svvlx4h, authState=authenticated
[2026-05-03T18:34:31.979681] [DEBUG] [AUTH] (AppRouter) Router redirect: location=/support-center, authState=authenticated
[2026-05-03T18:34:31.995744] [INFO] (ScreenAnalytics) 📱 Screen load started: support-center
[2026-05-03T18:34:31.995747] [DEBUG] [UI] (PageLoadObserver) Page push tracked: support-center
[2026-05-03T18:34:31.995755] [DEBUG] [VIDEO] (VideoControllerCleanup) 🧹 Invalidating all video controllers
[2026-05-03T18:34:31.995761] [INFO] [SYSTEM] (VideoStopNavigatorObserver) 📱 Navigation didPush to route: support-center - stopped all videos
[2026-05-03T18:34:32.003858] [INFO] (ScreenAnalytics) ✅ Screen content visible: support-center in 8ms
[2026-05-03T18:34:32.003889] [INFO] (PagePerf) PERF: support-center — visible: 8ms
[2026-05-03T18:34:36.456071] [DEBUG] [SYSTEM] (VideoStopNavigatorObserver) 📱 Skipping video disposal for popup route: null
[2026-05-03T18:35:59.448128] [INFO] [SYSTEM] Collecting bug report diagnostics
```

### Expected Result

(Generated from Zendesk Ticket: rabblelabs.zendesk.com/agent/tickets/1563)

### PR DESCRIPTION
**Pull request by untreu2**
## Description
After a user edited their video (e.g. changing hashtags), two bugs caused engagement counts to drop to zero:

1. _mergeUpdatedVideo (previously _preserveOriginalTimestamp) now carries nostrLikeCount forward from the existing video when the incoming update has null -- which is always the case after VideoEvent.fromNostrEvent(), since that field is runtime-injected and never stored in Nostr tags. The relay-fetched count will overwrite it once the next batch fires, so there is no risk of a stale count persisting.

2. _updateVideo() in share_video_menu now re-adds loops/likes/reposts/views tags from the original nostrEventTags when rebuilding the replacement event. These tags are baked into Vine-imported video events and were being silently dropped on every edit, permanently zeroing originalLoops and originalLikes.

Also renames _preserveOriginalTimestamp to _mergeUpdatedVideo to reflect the broader responsibility of the function.

**Related Issue:** Closes #3923

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by rabble** [comment]
**[real-bug]** This PR fixes a real issue where metadata edits (e.g. adding hashtags) cause engagement metrics to be lost. The changes rename `_preserveOriginalTimestamp` to `_mergeUpdatedVideo` and extend it to preserve runtime-injected like counts, plus add tag preservation logic for Vine-imported engagement counts. The implementation directly touches the merge logic and includes comprehensive test coverage for both the service layer and widget layer. This appears to be a legitimate bug fix with appropriate defensive measures.

<details><summary>signals</summary>

- diff modifies `_preserveOriginalTimestamp` method to also preserve `nostrLikeCount` with detailed comment explaining the runtime-injection and relay-fetch pattern
- engagement count tag preservation added to `_EditVideoDialogState._updateVideo()` to prevent loops/likes/reposts/views from being dropped during metadata edits
- test added `updateVideoEvent preserves nostrLikeCount from existing video` that reproduces the scenario: inject likes, edit hashtags, verify likes survive
- test added `engagement count tag preservation` suite with 6 test cases covering loops, likes, reposts, views extraction and edge cases
- PR is 0 commits behind main, no stale-base concern
</details>

_— divine-brain pr-classifier_

**Review by rabble — COMMENTED** [review]
## Review of PR #4150 — preserve like counts and engagement tags after metadata edit

Closes #3923. Two related fixes; cleanly motivated, correct in production code, and the rename `_preserveOriginalTimestamp → _mergeUpdatedVideo` is justified by the broadened responsibility.

### Overall

- **`_mergeUpdatedVideo` carry-forward of `nostrLikeCount`** — correct. Confirmed against `VideoEvent.copyWith` at `mobile/packages/models/lib/src/video_event.dart:1362`: it uses the `??` pattern (`nostrLikeCount ?? this.nostrLikeCount`), so passing a non-null value sets it. ✅
- **`share_video_menu.dart` engagement-tag preservation** — correct in shape; also confirmed to be the *only* runtime-injected count field (no `nostrCommentCount` / `nostrRepostCount` etc. in the model).
- **Comment block in `_mergeUpdatedVideo`** — 6 lines is at the upper edge of the project's "2–3 lines max" rule but the explanation is genuinely critical (non-obvious null-vs-runtime-injection invariant), so I'd leave it.

### 🛑 Blocking — widget tests don't actually test production code

`mobile/test/widgets/share_video_menu_edit_imeta_test.dart` defines a **standalone helper** `_extractEngagementTags` at the *bottom of the test file* and the new tests call only that helper:

```dart
const _engagementCountTagNames = {'loops', 'likes', 'reposts', 'views'};

List<List<String>> _extractEngagementTags(List<List<String>> nostrEventTags) {
  return nostrEventTags
      .where(
        (tag) =>
            tag.isNotEmpty &&
            _engagementCountTagNames.contains(tag[0]) &&
            tag.length >= 2,
      )
      .toList();
}
```

This is a verbatim copy of the production filter inside `_EditVideoDialog._updateVideo`. The five tests then assert the helper's behavior — which is the **test's own logic, not the production code**. If someone later changes the production filter (drops `length >= 2`, adds a new engagement tag, accidentally hardcodes a typo, etc.), the helper still passes the tests. The protection these tests advertise is illusory.

Two ways out, in order of preference:
1. **Pump `_EditVideoDialog`**, drive the form, intercept the `createAndSignEvent` call (or whatever publishes the replacement), and assert the captured tags include the engagement entries from `widget.video.nostrEventTags`. This is the only test that catches a regression in the actual production filter.
2. **Or extract the filter into a top-level / static method** in `share_video_menu.dart` (e.g. `extractEngagementTags(List<List<String>>) → List<List<String>>`), call it from both production and tests, and delete the duplicate helper. That way the unit tests at least exercise the same code production runs.

Option 1 better matches the test's intent ("the replacement event preserves engagement tags"), but Option 2 is a low-cost upgrade if Option 1 is too much widget-test plumbing.

### Should-fix

- **Redundant condition** in both production and test helper:
  ```dart
  if (tag.isNotEmpty && ... && tag.length >= 2)
  ```
  `tag.length >= 2` already implies `tag.isNotEmpty`. Drop the first clause.

- **Service test does exercise real code path** ✅ — `service.handleEventForTesting` → `_mergeUpdatedVideo` is real. The new test there is solid.

### Nits

- Comment in `share_video_menu.dart` says *"zeroing originalLoops/originalLikes forever"* — strictly speaking, until the next published event with the tags reappears (which won't happen via this UI flow). The comment is accurate enough.
- The `engagementCountTagNames` constant is a `const` set inside the function. If it's used in two places (it isn't yet, but might be), lift to a file-level const so production and tests share it.

### Performance / security

- No new I/O, no new auth boundaries, no new external calls. The fix is a single-pass list filter on a small tag list — performance non-issue.

### Verdict

**Request changes** for the widget-test concern only. Once those tests actually exercise `_updateVideo` (or share the filter via an extracted helper), this is a clean ship.

**Comment by rabble** [comment]
Pushed a follow-up commit `096acb61b` that addresses the blocking review point.

**What changed**

- Extracted the engagement-tag filter into a top-level `extractEngagementCountTags(List<List<String>>)` in `mobile/lib/widgets/share_video_menu.dart`, gated on `@visibleForTesting` (mirroring the existing `sendPostPublishCollaboratorInvites` pattern in the same file).
- `_EditVideoDialogState._updateVideo` now calls that function — the inline filter is gone.
- The widget tests in `share_video_menu_edit_imeta_test.dart` import and call the production function directly. The duplicate `_extractEngagementTags` helper and `_engagementCountTagNames` constant at the bottom of the test file are deleted.
- Also dropped the redundant `tag.isNotEmpty` clause since `tag.length >= 2` already implies non-empty.

The 5 engagement-tag tests now exercise the actual production code; a regression in the filter would surface immediately.

**Verification**

- `flutter analyze lib/widgets/share_video_menu.dart test/widgets/share_video_menu_edit_imeta_test.dart` — clean
- `flutter test test/widgets/share_video_menu_edit_imeta_test.dart` — 23 pass (was 22 plus a duplicate-helper test that disappeared with the helper)
- `flutter test test/services/video_event_service_replaceable_test.dart` — 9 pass (the new `nostrLikeCount` preservation test still green)

The other items (redundant `tag.isNotEmpty`) are also addressed by the same change. Ready for re-review.

File: mobile/lib/widgets/share_video_menu.dart [hunk 1]
```diff
@@ -80,6 +80,27 @@ sendPostPublishCollaboratorInvites({
   return results;
 }
 
+/// Tag names that carry engagement counts on Vine-imported video events.
+///
+/// These are baked into the Nostr event body and must survive a metadata
+/// edit; without that, `originalLoops` / `originalLikes` get permanently
+/// zeroed on the replacement event. See [extractEngagementCountTags].
+const _engagementCountTagNames = {'loops', 'likes', 'reposts', 'views'};
```
Comment (NotThatKindOfDrLiz) [inline mobile/lib/widgets/share_video_menu.dart]: This preservation set is still incomplete. `VideoEvent.fromNostrEvent()` also parses `comments` into `originalComments`, and the repo hydration paths treat `originalComments == null` as missing stats. As written, edited Vine imports can still lose comment counts after republish. Please add `comments` here and extend the test coverage to match.

File: mobile/lib/services/video_event_service.dart [hunk 2]
```diff
@@ -4082,17 +4082,30 @@ class VideoEventService extends ChangeNotifier implements VideoEventCache {
 
   /// Preserve original timestamp when updating video events
```
Comment (NotThatKindOfDrLiz) [inline mobile/lib/services/video_event_service.dart]: The rename is correct, but this dartdoc is now stale: `_mergeUpdatedVideo()` preserves more than timestamps. Please update the docs to describe the full merge contract here, including the runtime `nostrLikeCount` carry-forward, so the comment matches the functions behavior.

File: mobile/test/widgets/share_video_menu_edit_imeta_test.dart [hunk 3]
```diff
@@ -233,4 +235,65 @@ void main() {
       expect(_extractImetaUrls(tags), isEmpty);
     });
   });
```
Comment (NotThatKindOfDrLiz) [inline mobile/test/widgets/share_video_menu_edit_imeta_test.dart:235]: These tests now exercise the extracted helper, which is better than the prior mirror-helper pattern, but they still do not verify the actual `_updateVideo()` publish path. Please add one test that captures the `tags` passed to `createAndSignEvent()` and asserts the preserved engagement tags survive a metadata edit. Otherwise a future refactor can drop the `tags.addAll(extractEngagementCountTags(...))` call and this suite will still pass.

**Review by NotThatKindOfDrLiz — CHANGES_REQUESTED** [review]
This fixes part of the bug, but Im requesting changes before merge because the current patch still leaves part of the engagement contract behind and the test coverage does not yet protect the real publish path.

Main blockers:
- `comments` is missing from the preserved engagement tags. `VideoEvent.fromNostrEvent()` parses it into `originalComments`, and downstream hydration treats `originalComments == null` as missing stats. Leaving it out means edited Vine imports can still lose comment counts after republish.
- `_mergeUpdatedVideo()` now preserves both timestamps and runtime like count, but the dartdoc above it still documents only the old timestamp-only behavior. That is stale immediately.
- The new widget test coverage exercises the extracted helper, not `_EditVideoDialogState._updateVideo()` / `createAndSignEvent(tags: ...)`. A future refactor can drop the `tags.addAll(extractEngagementCountTags(...))` call and these tests will still pass.

What Id like to see here:
- Add `comments` to the preserved engagement tag set and extend the assertions accordingly.
- Update `_mergeUpdatedVideo()` docs so they match the functions actual responsibilities.
- Add one end-to-end test on the edit flow that captures the tags passed to `createAndSignEvent()` and proves the preserved engagement tags survive a metadata edit.

Once those are in, the fix looks directionally right.

**Comment by untreu2** [comment]
Addressed comments @NotThatKindOfDrLiz

**Review by NotThatKindOfDrLiz — APPROVED** [review]
Rereviewed final head 17d9188242c95d76354c973e93f1e07c7522609a after debt cleanup. Checks are green and I have no remaining findings.
